sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -0,0 +1,665 @@
+  properties<PropertyListV> P) noexcept;
+----
+|
+Constructs an `annotated_ptr` object. Does not allocate new storage. The
+underlying pointer is initialized with `Ptr`. `PropertyListU` and
+`PropertyListV` will be combined to construct `PropertyListT`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr(annotated_ptr const &) noexcept = default;
+----
+|
+Constructs an `annotated_ptr` object from another `annotated_ptr` with the same
+template parameterization object.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+reference operator*() const;
+----
+|
+Returns an `annotated_ref` reference wrapper which can be used to read or write
+to the underlying pointer. Reads/Writes using the reference will retain the
+annotations.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+reference operator[](std::ptrdiff_t Index) const;
+----
+|
+Returns an `annotated_ref` reference wrapper to the object at offset `Index`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr operator+(size_t Offset) const;
+----
+|
+Returns an `annotated_ptr` that points to `this[Offset]`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+std::ptrdiff_t operator-(annotated_ptr FromPtr) const;
+----
+|
+Returns the distance between the underlying pointer and the pointer encapsulated
+by `FromPtr`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+operator bool() const noexcept;
+----
+|
+Returns `false` if the underlying pointer is null, returns `true` otherwise.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+/*unspecified*/ operator T*() noexcept = delete;
+/*unspecified*/ operator const T*() const noexcept = delete;
+----
+|
+Implicit conversion to a pointer to the underlying type is not supported.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+T* get() noexcept;
+const T* get() const noexcept;
+----
+|
+Returns the underlying raw pointer. The raw pointer will not retain the
+annotations.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator=(const T*) noexcept;
+----
+|
+Allows assignment from a pointer to type `T`.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator=(annotated_ptr const&) noexcept;
+----
+|
+Allows assignment from an `annotated_ptr` with the same parameterization.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator++() noexcept;
+----
+|
+Prefix increment operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr operator++(int) noexcept;
+----
+|
+Postfix increment operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr& operator--() noexcept;
+----
+|
+Prefix decrement operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+annotated_ptr operator--(int) noexcept;
+----
+|
+Postfix decrement operator.
+
+// --- ROW BREAK ---
+a|
+[source,c++]
+----
+template<typename propertyT>
+static constexpr bool has_property();
+----
+| 
+Returns true if the property list contains the property with property key class
```

Comment:
"property list" is no longer the official name and the template argument to this function is named "propertyT" so I'd reword this as:

```suggestion
Returns true if `PropertyListT` contains the property with `propertyT` key class.  
```

Reply:
Can you make this same change in the annotated_arg spec?